File: backend/todo/src/main/java/com/msante/todo/services/TodoService.java
package com.msante.todo.services;

import com.msante.todo.entities.TodoEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface TodoService {
    TodoEntity createTask(TodoEntity task);
    TodoEntity updateTask(TodoEntity task);
    void deleteTask(Long id);
    TodoEntity getTaskById(Long id);
    List<TodoEntity> getAllTasks();

}

File: frontend/todo/src/app/services/todo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Task } from '../modals/task';

@Injectable({
  providedIn: 'root'
})
export class TodoService {

  constructor(private httpClient:HttpClient) { }

   _url="http://localhost:8080/task";

  getTaskById(id:number){
    return this.httpClient.get(this._url+"/"+id);
  }
  getAllTasks(){
    return this.httpClient.get(this._url+"/all");
  }
  createTask(task:Task){
    return this.httpClient.post(this._url,task);
  }
  updateTask(task:Task){
    return this.httpClient.put(this._url,task);
  }
  deleteTask(id:number){
    return this.httpClient.delete(this._url+"/"+id);
  }
}
